### ISSUE TITLE
Allow extents to be set for XYZ tile layers

### ISSUE THREAD

**Issue by Wardrop** [qgis/QGIS#33834]
**Feature description.**
It would be handy if _extents_ could be specified for XYZ tile layers. The XYZ coordinates would then be relative to these extents. At the moment, the extents default to the globe, so QGIS treats the XYZ coordinate as relative to the globe, where as some tile servers serve tiles relative to extents of the dataset they're serving.

**Additional context**
We use a GIS solution that uses OpenLayers and MapServer. The service dynamically generates tiles on the fly based on the XYZ standard. Being a TMS-based service, the Y axis is inverted, so I had to do the "{-y}" trick to get the tiles to display correctly. The problem I then encountered was the scale and position were way out. I worked out that our tile server was serving the tiles relative to the extents defined on the tile server. I believe this is done so as to not waste time generating and caching tiles that are outside those extents.

To workaround the problem and confirm my suspicions, I configured a new tile service on the server with the extents set to that of the globe. The tiles then rendered at the correct scale and position in QGIS. The problem being that now I have two tile sets serving the same data but with different coordinate XYZ coordinates. It would be handy if in QGIS one could define the extents of the XYZ layer, and if QGIS would reference the XYZ coordinate relatively to those extents.

Hopefully that description of the problem is adequate.

**Comment by wonder-sk** [comment]
Maybe a better solution would be to have TMS protocol supported by QGIS, so you don't even need to enter the extent manually when defining the service?

By the way, MapServer should also support WMTS which I understand is a more standard replacement for TMS - maybe it's easier to switch to that?

**Comment by Wardrop** [comment]
Does TMS have an endpoint that describes the layer (e.g. extents, zoom levels, etc)?

**Comment by wonder-sk** [comment]
Yes it does, see https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification

**Comment by hxcode** [comment]
You can new a bookmark which fill extents, then click bookmark in browser panel to zoom.
